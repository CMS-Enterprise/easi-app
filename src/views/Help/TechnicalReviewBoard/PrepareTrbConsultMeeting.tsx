import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Link, SummaryBox } from '@trussworks/react-uswds';

import HelpBreadcrumb from 'components/HelpBreadcrumb';
import HelpPageIntro from 'components/HelpPageIntro';
import MainContent from 'components/MainContent';
import RelatedArticles from 'components/RelatedArticles';
import { CMS_TRB_EMAIL, TRB_DECK_TEMPLATES } from 'constants/externalUrls';
import useScrollToLocation from 'hooks/useScrollToLocation';

import NeedHelpBox from '../InfoBox/NeedHelpBox';

function PrepareTrbConsultMeeting() {
  const { t } = useTranslation('prepareTrbConsultMeeting');

  useScrollToLocation();

  return (
    <>
      <MainContent className="grid-container margin-bottom-7">
        <HelpBreadcrumb type="Close tab" />
        <HelpPageIntro
          type="Technical Review Board"
          heading={t('title')}
          subheading={t('description')}
        />

        <h2 className="font-heading-xl line-height-heading-2 margin-top-3 margin-bottom-2">
          {t('what.title')}
        </h2>
        <p className="line-height-body-5">{t('what.description')}</p>

        <h3 className="margin-top-3 margin-bottom-1">
          {t('whatToExpect.title')}
        </h3>
        <div className="line-height-body-5">
          {t('whatToExpect.description')}
        </div>

        <div className="bg-base-lightest margin-top-2 padding-2">
          <h3 className="margin-top-0 margin-bottom-1">{t('tips.title')}</h3>
          <ul className="easi-list line-height-body-5">
            <li>{t('tips.list.0')}</li>
            <li>{t('tips.list.1')}</li>
            <li>{t('tips.list.2')}</li>
            <li>{t('tips.list.3')}</li>
            <li>{t('tips.list.4')}</li>
            <li>{t('tips.list.5')}</li>
          </ul>
        </div>

        <h2 className="margin-top-6 margin-bottom-2">
          {t('whatToBring.title')}
        </h2>
        <div className="line-height-body-5 margin-bottom-1">
          {t('whatToBring.description')}
        </div>
        <ul className="easi-list line-height-body-5">
          <li>{t('whatToBring.list.0')}</li>
          <li>{t('whatToBring.list.1')}</li>
          <li>{t('whatToBring.list.2')}</li>
        </ul>

        <div id="download-presentation-templates" />
        <SummaryBox
          heading={t('downloadTemplates.title')}
          className="desktop:grid-col-6 margin-top-4"
        >
          <p>
            <Trans
              i18nKey="prepareTrbConsultMeeting:downloadTemplates.description"
              components={{
                a: <Link href={`mailto:${CMS_TRB_EMAIL}`}> </Link>,
                email: CMS_TRB_EMAIL
              }}
            />
          </p>
          <div>
            <Link href={TRB_DECK_TEMPLATES} variant="external">
              {t('downloadTemplates.link')}
            </Link>
          </div>
        </SummaryBox>

        <h2 className="margin-top-6 margin-bottom-2">{t('outcomes.title')}</h2>
        <div className="line-height-body-5 margin-bottom-1">
          {t('outcomes.description')}
        </div>
        <ul className="easi-list line-height-body-5">
          <li>{t('outcomes.list.0')}</li>
          <li>{t('outcomes.list.1')}</li>
          <li>{t('outcomes.list.2')}</li>
          <li>{t('outcomes.list.3')}</li>
        </ul>

        <NeedHelpBox
          className="desktop:grid-col-6 margin-top-6"
          content={t('help.text')}
          email={CMS_TRB_EMAIL}
        />
      </MainContent>

      <RelatedArticles
        type="Technical Review Board"
        currentArticle="prepareTrbConsultMeeting"
      />
    </>
  );
}

export default PrepareTrbConsultMeeting;
