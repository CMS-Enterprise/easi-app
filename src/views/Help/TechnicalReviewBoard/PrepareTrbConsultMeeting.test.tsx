import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import PrepareTrbConsultMeeting from './PrepareTrbConsultMeeting';

describe('Trb Help Prepare Consult Meeting', () => {
  it('renders', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PrepareTrbConsultMeeting />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
