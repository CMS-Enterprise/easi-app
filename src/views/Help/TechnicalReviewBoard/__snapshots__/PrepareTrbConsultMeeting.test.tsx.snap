// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trb Help Prepare Consult Meeting renders 1`] = `
<DocumentFragment>
  <main
    class="easi-main-content grid-container margin-bottom-7"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <button
      class="usa-button usa-button--unstyled margin-top-6"
      data-testid="button"
      type="button"
    >
      <svg
        class="usa-icon margin-right-05 margin-top-3px text-tbottom"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      Close tab
    </button>
    <div
      class="help-page-intro margin-bottom-4"
    >
      <h1
        aria-live="polite"
        class="easi-h1 margin-bottom-0 margin-top-3"
        tabindex="-1"
      >
        Prepare for your TRB consult meeting
      </h1>
      <a
        class="usa-link display-inline-block margin-y-1"
        href="/help/trb"
      >
        <span
          class="easi-tag system-profile__tag text-primary-dark bg-primary-lighter padding-bottom-1"
          data-testid="tag"
        >
          Technical Review Board
        </span>
      </a>
      <div
        class="font-body-lg margin-top-1 margin-bottom-0 line-height-body-5"
      >
        Learn what to expect during and after the meeting, including what to prepare for and what to bring.
      </div>
    </div>
    <h2
      class="font-heading-xl line-height-heading-2 margin-top-3 margin-bottom-2"
    >
      What is the Technical Review Board?
    </h2>
    <p
      class="line-height-body-5"
    >
      The CMS Technical Review Board (TRB) is a technical assistance resource for project teams across the agency at all stages of their system’s life cycle. It offers consultations and reviews on an ongoing or one-off basis, allowing project teams to consult with a cross-functional team of technical advisors. It also guides project teams on adhering to CMS technical standards and leveraging existing technologies.
    </p>
    <h3
      class="margin-top-3 margin-bottom-1"
    >
      What to expect at the meeting
    </h3>
    <div
      class="line-height-body-5"
    >
      TRB consultation meetings are there to help the TRB understand what your problem or idea is, and details surrounding any proposed solution. You can expect the TRB and any additional Subject Matter Experts (SMEs) to ask many questions about your system, problem, and solution.
    </div>
    <div
      class="bg-base-lightest margin-top-2 padding-2"
    >
      <h3
        class="margin-top-0 margin-bottom-1"
      >
        Tips for the meeting
      </h3>
      <ul
        class="easi-list line-height-body-5"
      >
        <li>
          Keep it small, but make sure to bring any additional attendees who can help explain your problem or solution and can answer questions from the TRB.
        </li>
        <li>
          A presentation deck is not necessary but is recommended. You can download TRB template decks below.
        </li>
        <li>
          If you have specific questions you want answered by the TRB, make sure to bring a list with you.
        </li>
        <li>
          Keep presentations brief, but thorough.
        </li>
        <li>
          Don’t worry about running out of time. You can always schedule follow-up sessions with the TRB
        </li>
        <li>
          If you have one, show your architecture diagram during your presentation. Above all else, this document helps the TRB understand your system, problem, and proposed solution.
        </li>
      </ul>
    </div>
    <h2
      class="margin-top-6 margin-bottom-2"
    >
      What to bring
    </h2>
    <div
      class="line-height-body-5 margin-bottom-1"
    >
      The following are important items to have with you at the TRB meeting:
    </div>
    <ul
      class="easi-list line-height-body-5"
    >
      <li>
        an architecture diagram if you have one
      </li>
      <li>
        a presentation slide deck if you choose to share one. You may download TRB deck templates below.
      </li>
      <li>
        any additional materials you wish to discuss at the meeting, such as a business case, incident report, security findings, etc.
      </li>
    </ul>
    <div
      id="download-presentation-templates"
    />
    <div
      class="usa-summary-box desktop:grid-col-6 margin-top-4"
      data-testid="summary-box"
    >
      <div
        class="usa-summary-box__body"
      >
        <h3
          class="usa-summary-box__heading"
        >
          Download TRB presentation deck templates
        </h3>
        <div
          class="usa-summary-box__text"
        >
          <p>
            If you have any questions about the templates or would like help filling one out, you may contact the TRB at 
            <a
              class="usa-link"
              href="mailto:cms-trb@cms.hhs.gov"
            >
              cms-trb@cms.hhs.gov
            </a>
            .
          </p>
          <div>
            <a
              class="usa-link usa-link--external"
              href="https://share.cms.gov/cms-wide/Systems/TRB/Templates/Forms/AllItems.aspx"
            >
              Download deck templates (zip)
            </a>
          </div>
        </div>
      </div>
    </div>
    <h2
      class="margin-top-6 margin-bottom-2"
    >
      Potential outcomes
    </h2>
    <div
      class="line-height-body-5 margin-bottom-1"
    >
      After the meeting, the TRB will deliberate and provide recommendations and next steps. Within 5 business days, the TRB will compile an advice letter for your project and send it to you. The advice letter may contain:
    </div>
    <ul
      class="easi-list line-height-body-5"
    >
      <li>
        recommendations for changes to your system or solution
      </li>
      <li>
        additional teams or points of contact to reach out to who may be able to help you further
      </li>
      <li>
        suggestions for systems, services, or solutions to use to implement your idea or solve your problem
      </li>
      <li>
        a request for a follow-up consultation session
      </li>
    </ul>
    <div
      class="usa-summary-box desktop:grid-col-6 margin-top-6"
      data-testid="summary-box"
    >
      <div
        class="usa-summary-box__body"
      >
        <h3
          class="usa-summary-box__heading"
        >
          Need help?
        </h3>
        <div
          class="usa-summary-box__text"
        >
          <div
            class="margin-top-2 margin-bottom-05"
          >
            Contact the Technical Review Board:
          </div>
          <div>
            <a
              class="usa-link"
              href="mailto:cms-trb@cms.hhs.gov"
            >
              cms-trb@cms.hhs.gov
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
  <div
    class="bg-base-lightest"
  >
    <main
      class="easi-main-content grid-container padding-y-2"
      id="main-content"
      role="main"
      tabindex="-1"
    >
      <h2
        class="margin-bottom-1"
      >
        Related help articles
      </h2>
      <dt
        class="margin-bottom-4"
      >
        Below are some additional help articles that you may find useful.
      </dt>
      <ul
        class="usa-card-group margin-y-2"
        data-testid="CardGroup"
      >
        <li
          class="usa-card desktop:grid-col-4 article article-card--isLink"
          data-testid="article-card"
        >
          <div
            class="usa-card__container radius-md shadow-2 minh-mobile"
          >
            <header
              class="usa-card__header padding-x-3 padding-top-3"
              data-testid="CardHeader"
            >
              <h3
                class="line-height-body-4 margin-bottom-1"
              >
                Steps involved in getting technical assistance from the TRB
              </h3>
              <a
                class="usa-link display-inline-block margin-bottom-1"
                href="/help/trb"
              >
                <span
                  class="easi-tag system-profile__tag text-primary-dark bg-primary-lighter padding-bottom-1"
                  data-testid="tag"
                >
                  Technical Review Board
                </span>
              </a>
            </header>
            <div
              class="usa-card__body padding-top-0 article__body"
              data-testid="CardBody"
            >
              <p>
                The CMS Technical Review Board (TRB) is a technical assistance resource for project teams across the agency at all stages of their system’s life cycle. It offers consultations and reviews on an ongoing or one-off basis, allowing project teams to consult with a cross-functional team of technical advisors. It also guides project teams on adhering to CMS technical standards and leveraging existing technologies.
              </p>
            </div>
            <div
              class="usa-card__footer padding-top-2 article__footer"
              data-testid="CardFooter"
            >
              <a
                class="usa-link easi-request__button-link padding-x-2"
                href="/"
              >
                Read
              </a>
            </div>
          </div>
        </li>
      </ul>
    </main>
  </div>
</DocumentFragment>
`;
